dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/CustomizationFilesServlet.java

```diff
@@ -164,7 +79,97 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
      */
     @Override
     public String getServletInfo() {
-        return "Short description";
-    }// </editor-fold>
+        return "CustomizationFilesServlet – for serving customizable parts of the page";
+    }
+
+    /**
+     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
+     * methods.
+     *
+     * @param request  servlet request
+     * @param response servlet response
+     */
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response) {
+        response.setContentType("text/html;charset=UTF-8");
+        String customFileType = request.getParameter("customFileType");
+
+        Optional<File> file = findFile(customFileType);
+        if (!file.isPresent()) {
+            return;
+        }
+
+        File fileIn = file.get();
+        try (FileInputStream inputStream = new FileInputStream(fileIn);
+             BufferedReader in = new BufferedReader(new InputStreamReader(inputStream));
+             PrintWriter out = response.getWriter()) {
+            in.lines()
+                    .forEach(out::println);
+        } catch (Exception e) {
+            // We don't want to send trash to logs, but nevertheless some info should be present on demand
+            logger.log(Level.FINEST, e, () -> "Servlet encountered exception: ");
+        }
+    }
+
+    // -------------------- PRIVATE --------------------
 
+    private Optional<File> findFile(String customFileType) {
+        return getPaths(customFileType).stream()
+                .map(Path::toFile)
+                .filter(f -> f.exists() && f.isFile())
+                .findFirst();
+    }
+
+    private List<Path> getPaths(String customFileType) {
+        Key key = SETTINGS_MAPPING.get(customFileType);
+        String basePath = getFilePath(key);
+        List<Path> paths = new ArrayList<>();
+        if (LOCALIZED_FILES_KEYS.contains(key)) {
+            Optional<Path> localizedPath = createLocalizedPath(basePath);
+            localizedPath.ifPresent(paths::add);
+        }
+        paths.add(Paths.get(basePath));
+        return paths;
+    }
+
+    private String getFilePath(Key key) {
+        return key != null ? settingsService.getValueForKey(key) : StringUtils.EMPTY;
+    }
+
+    private Optional<Path> createLocalizedPath(String basePath) {
+        if (StringUtils.isBlank(basePath) || !basePath.contains(".")) {
+            return Optional.empty();
+        }
+        return Optional.of(Paths.get(interpolateLocaleCodeIntoPath(basePath, obtainLocaleCode())));
```

Comment:
We could rewrite it to make it more functional-like with:
`        return Optional.ofNullable(basePath)
                .filter(path -> basePath.contains("."))
                .map(path -> Paths.get(interpolateLocaleCodeIntoPath(path, obtainLocaleCode())));
`